### ISSUE TITLE
MNIST Example runs train loader twice per epoch?

### ISSUE THREAD

**Issue by n2cholas** [pytorch/ignite#1059]
I'm new to ignite, and was looking at the MNIST example. At the end of each training epoch, the evaluator is run on the train loader (https://github.com/pytorch/ignite/blob/master/examples/mnist/mnist.py#L78). This means the train loader is run twice per epoch, right? Doesn't it make more sense to compute a running average (using a Metric) throughout the training epoch? Or report the running average of the training metrics every few batches? Thanks in advance.

**Comment by sdesrozis** [comment]
@n2cholas Thanks for the question.

Yes, training set is used twice per epoch. The first time is to train the model. The second, cheaper, is to compute metrics (accuracy, f1, etc.) and evaluate how performs the model on data used for training. Sometimes, you will see people who compute metrics during the training phase, meaning the model is changing. A good practice is to do it when model is fixed. 

To evaluate how could be the generalization, the eval dataset is used.  

HTH

**Comment by n2cholas** [comment]
Thanks for the quick reply @sdesrozis!
